parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -125,6 +127,16 @@ public static WriteBuilder write(OutputFile file) {
     return new WriteBuilder(file);
   }
 
+  public static WriteBuilder write(EncryptedOutputFile file) {
+    if (EncryptionUtil.useNativeEncryption(file.keyMetadata())) {
+      return write(file.rawOutputFile())
+          .withFileEncryptionKey(file.keyMetadata().encryptionKey())
```

Comment:
I think I suggested adding `encryptionKey` and `aadPrefix` to the base, but after looking at the `castOrParse`, I think it would be cleaner to use that here to get an instance of `StandardKeyMetadata`.

Reply:
I think this also needs to call `withKeyMetadata` so that the key metadata instance is set.

After that, it's awkward that the key metadata is set separately than the encryption key and AAD prefix in this case because the key metadata is not opaque. I think the solution is to do the `instanceof` check above in `withKeyMetadata`. That way if the key metadata is `StandardKeyMetadata` then it will also configure the AAD prefix and key. If not, then setting them separately makes sense.

Reply:
+1 to the first suggestion.

As for the second - this `WriterBuilder` class doesn't have a `withKeyMetadata` method (since it is a basic writer appender).